src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -624,40 +624,45 @@ private bool TryFindFreeIndex(out int index)
             }
         }
 
+        //cache ClassIndex and ProcdureNode for repeated calls of the same type object.
+        private int previousClassIndex;
+        private ProcedureNode pn;
```

Comment:
I think we should rename this to something more descriptive for clarity.

Reply:
I renamed it to `disposeProcedureNode`